src/styles/main.scss

```
@@ -0,0 +1,61 @@
+* {
+  box-sizing: border-box;
+}
+
+body {
+  height: 100vh;
+  margin: 0;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+}
+
+.calendar {
+  display: flex;
+  width: 726px;
+  flex-wrap: wrap;
+  gap: 1px;
+  padding: 10px;
+
+  &__day {
+    border: 1px solid black;
+    height: 100px;
+    width: 100px;
+    background-color: #eee;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+  }
+
+  &__day:hover {
+    cursor: pointer;
+    background-color: #ffbfcb;
+    transform: translateY(-20px);
+    transition-duration: 0.5s;
+  }
+
+  @for $num from 1 through 31 {
+    &__day:nth-child(#{$num})::before {
+      content: '#{$num}';
+    }
+  }
+}
+
+$day: (mon 0, tue 101, wed 202, thu 303, fri 404, sat 505, sun 606);
+
+@each $days, $margin in $day {
+  .calendar--start-day-#{$days} .calendar__day:first-child {
+    margin-left: $margin;
+  }
+}
+
+$lenghts: (28 29, 29 30, 30 31, 31 32);
```

Remove duplicates and month length cannot be 32. Also, move this variable to the separate file
```suggestion
$lenghts: (28, 29, 30, 31);
```